tests/v2/provisioining/provisioining_harvester_node_driver_test.go

```diff
@@ -0,0 +1,222 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RKE2NodeDriverProvisioningTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	standardUserClient *rancher.Client
+}
+
+const (
+	baseHarvesterClusterName = "qaauto-harvester-cluster"
+	namespace                = "fleet-default"
+	defaultRandStringLength  = 5
+)
+
+func GenerateRandomName(baseClusterName string) string {
+	clusterName := baseClusterName + namegenerator.RandStringLower(5)
+	return clusterName
+}
+
+func (r *RKE2NodeDriverProvisioningTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *RKE2NodeDriverProvisioningTestSuite) SetupSuite() {
+	testSession := session.NewSession(r.T())
+	r.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+
+	r.client = client
+
+	enabled := true
+	user := &management.User{
+		Username: GenerateRandomName("user"),
+		Password: GenerateRandomName("password"),
+		Name:     "qatest-user",
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+}
+
+func (r *RKE2NodeDriverProvisioningTestSuite) TestProvisioning_RKE2HarvesterCluster() {
+	subSession := r.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := r.client.WithSession(subSession)
+	require.NoError(r.T(), err)
+
+	cloudCredential, err := harvester.CreateHarvesterCloudCredentials(client)
+	require.NoError(r.T(), err)
+
+	nodeRoles0 := []map[string]bool{
+		{
+			"controlplane": true,
+			"etcd":         true,
+			"worker":       true,
+		},
+	}
+
+	nodeRoles1 := []map[string]bool{
+		{
+			"controlplane": true,
+			"etcd":         false,
+			"worker":       false,
+		},
+		{
+			"controlplane": false,
+			"etcd":         true,
+			"worker":       false,
+		},
+		{
+			"controlplane": false,
+			"etcd":         false,
+			"worker":       true,
+		},
+	}
+
+	tests := []struct {
+		name      string
+		nodeRoles []map[string]bool
+		client    *rancher.Client
+	}{
+		{"1 Node all roles Admin User", nodeRoles0, r.client},
+		{"1 Node all roles Standard User", nodeRoles0, r.standardUserClient},
+		{"3 nodes - 1 role per node Admin User", nodeRoles1, r.client},
+		{"3 nodes - 1 role per node Standard User", nodeRoles1, r.standardUserClient},
+	}
+
+	for _, tt := range tests {
+		r.Run(tt.name, func() {
+			testSession := session.NewSession(r.T())
+			defer testSession.Cleanup()
+
+			testSessionClient, err := tt.client.WithSession(testSession)
+			require.NoError(r.T(), err)
+
+			clusterName := GenerateRandomName(baseHarvesterClusterName)
+			generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+			machinePoolConfig := machinepools.NewHarvesterMachineConfig(generatedPoolName, "default", "default/ctw-network-1", "default/image-rpj98")
```

Comment:
Can we parameterize these too. ?

Reply:
These could be a part of the config file.